Makefile

```diff
@@ -150,3 +147,20 @@ node_modules/.uptodate: package.json
 	@echo installing javascript dependencies
 	@node_modules/.bin/check-dependencies 2>/dev/null || npm install
 	@touch $@
+
+PYTHON_VERSION = 3.6.8
+PYTHON_MINOR_VERSION := $(shell echo $$PYTHON_VERSION | cut -d '.' -f1,2)
+PYTHON_VERSION_INSTALLED := $(shell python$${PYTHON_MINOR_VERSION} --version 2>/dev/null)
+.PHONY: python
+python:
+ifndef PYTHON_VERSION_INSTALLED
+	pyenv install ${PYTHON_VERSION}
+	pip install --upgrade pip
+endif
```

Comment:
If there's no `python3.6` on the $PATH then we try to use pyenv to install Python 3.6.8 (and crash if pyenv isn't installed or if it fails).

We don't insist on 3.6.8 -- if a different patch version of 3.6 is installed we'll use it and not try to install 3.6.8. This may not be ideal but some environments (e.g. Travis) just let you install "3.6" and don't let you specify the patch version (and currently Travis's 3.6 is 3.6.7). So we can't insist on 3.6.8. At least not until we get rid of Travis. In practice the exact patch version in dev and CI probably doesn't matter much.

Also note that we don't insist that Python 3.6 was installed with pyenv, or even that pyenv is installed. As long as Python 3.6 is on $PATH we will use it and not try to use pyenv. This is what happens in CI for example.

Reply:
For the record, this is some `Makefile` code that would test that Python 3.6.8 (exact patch version) is installed and install that even if a different Python 3.6.x is installed. It doesn't work on Travis:

```
PYTHON_VERSION = 3.6.8
PYTHON_MINOR_VERSION := $(shell echo $$PYTHON_VERSION | cut -d '.' -f1,2)
.PHONY: python
python:
	@if [ `python${PYTHON_MINOR_VERSION} --version 2>/dev/null | grep -c "${PYTHON_VERSION}"` -ne 1 ]; then pyenv install -s ${PYTHON_VERSION}; fi
```